src/sentry/integrations/api/serializers/rest_framework/data_forwarder.py

```diff
@@ -0,0 +1,206 @@
+        if not isinstance(write_key, str) or not re.match(segment_write_key_pattern, write_key):
+            raise ValidationError("write_key must be a valid Segment write key format")
+
+        return config
+
+    def _validate_splunk_config(self, config: dict) -> dict:
+        required_fields = ["instance_URL", "index", "source", "token"]
+
+        missing_fields = [field for field in required_fields if field not in config]
+        if missing_fields:
+            raise ValidationError(f"Missing required Splunk fields: {', '.join(missing_fields)}")
+
+        errors = []
+
+        splunk_url_pattern = r"^https?://[a-zA-Z0-9\-\.]+(?::\d+)?(?:/.*)?$"
+        instance_url = config.get("instance_URL")
+        if not isinstance(instance_url, str) or not re.match(splunk_url_pattern, instance_url):
+            errors.append("instance_URL must be a valid URL starting with http:// or https://")
+
+        index = config.get("index")
+        if not isinstance(index, str) or index.strip() == "":
+            errors.append("index must be a non-empty string")
+
+        source = config.get("source")
+        if not isinstance(source, str) or source.strip() == "":
+            errors.append("source must be a non-empty string")
+
+        token = config.get("token")
+        if not isinstance(token, str) or token.strip() == "":
+            errors.append("token must be a non-empty string")
+        else:
+            splunk_token_pattern = r"^[a-zA-Z0-9\-]+$"
+            if not re.match(splunk_token_pattern, token):
+                errors.append("token must be a valid Splunk HEC token format")
+
+        if errors:
+            raise ValidationError(errors)
+
+        return config
+
+    def validate(self, attrs: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        organization_id = attrs.get("organization_id")
+        provider = attrs.get("provider")
+
+        if organization_id and provider:
+            existing = DataForwarder.objects.filter(
+                organization_id=organization_id, provider=provider
+            )
+
+            if self.instance:
+                existing = existing.exclude(id=self.instance.id)
+
+            if existing.exists():
+                raise ValidationError(
+                    f"A DataForwarder with provider '{provider}' already exists for this organization."
+                )
+
+        return attrs
+
+
+class DataForwarderProjectSerializer(Serializer):
+    data_forwarder_id = serializers.IntegerField()
+    project_id = serializers.IntegerField()
+    overrides = serializers.JSONField(default=dict)
+    is_enabled = serializers.BooleanField(default=True)
+
+    def validate_data_forwarder_id(self, value: int) -> int:
+        try:
+            DataForwarder.objects.get(id=value)
+        except DataForwarder.DoesNotExist:
+            raise ValidationError("DataForwarder with this ID does not exist")
+        return value
+
+    def validate_project_id(self, value: int) -> int:
```

Comment:
er, leaving this comment more for me, but there may be some security concerns here for an inbound project serializer. The concern is basically, it's generally not enough to check that the Project exists. We also need to check the authenticated user/organization is also the owner of the Project. Again, I need to look up how we usually handle project id ownership validation...

But also in general, I think we should also include the `organization_id` whenever possible in the object lookups to ensure we're operating on something owned by the right org.